cou/apps/auxiliary.py

```diff
@@ -16,89 +16,62 @@
-        if self.origin_setting is None:
-            return None
-
-        # Ex: "cloud:focal-ussuri" will result in "ussuri"
-        os_origin_parsed: Optional[str] = self.os_origin.rsplit("-", maxsplit=1)[-1]
-        if os_origin_parsed == "distro":
-            # find the OpenStack release based on ubuntu series
-            os_origin_parsed = LTS_SERIES[self.series]
-        elif os_origin_parsed == "":
-            # if it's empty we consider the previous release from the target.
-            # Ex: rabbitmq-server has empty "source" and receive target "victoria".
-            # In that case it will be considered as ussuri and with the upgrade,
-            # "source" config will be changed to "cloud:focal-victoria".
-            os_origin_parsed = target.previous_release
-
-        return OpenStackRelease(os_origin_parsed) if os_origin_parsed else None
-
     @property
     def expected_current_channel(self) -> str:
         """Return the expected current channel based on the current OpenStack release.
 
         Note that this is not necessarily equal to the "channel" property since it is
         determined based on the workload version.
 
+        :raises ApplicationError: When cannot find a track.
         :return: The expected current channel for the application.
         :rtype: str
         """
-        return f"{self.openstack_to_track(self.current_os_release)}/stable"
+        track = openstack_to_track(self.charm, self.series, self.current_os_release)
+        if track:
+            return f"{track}/stable"
+
+        raise ApplicationError(
+            f"Cannot find a track of '{self.charm}' for {self.current_os_release.codename}"
+        )
 
     def target_channel(self, target: OpenStackRelease) -> str:
         """Return the channel based on the target passed.
 
         :param target: OpenStack release as target to upgrade.
         :type target: OpenStackRelease
-        :return: The next channel for the application. E.g: victoria/stable
+        :raises ApplicationError: When cannot find a track.
+        :return: The next channel for the application. E.g: 3.8/stable
         :rtype: str
         """
-        return f"{self.openstack_to_track(target)}/stable"
+        track = openstack_to_track(self.charm, self.series, target)
+        if track:
+            return f"{track}/stable"
+
+        raise ApplicationError(f"Cannot find a track of '{self.charm}' for {target.codename}")
 
     def _get_refresh_charm_plan(
         self, target: OpenStackRelease, parallel: bool = False
     ) -> Optional[UpgradeStep]:
         """Get plan for refreshing the current channel.
 
         This function also identifies if charm comes from charmstore and in that case,
-        makes the migration.
+        makes the migration. This method overwrite OpenStackApplication method because
+        a channel track from auxiliary charms can have multiple OpenStack releases.
```

Comment:
```suggestion
        makes the migration. This method overwrites the corresponding method from the  OpenStackApplication superclass because channels of "auxiliary" charms can be compatible with multiple OpenStack releases.
```